compiler/lib/src/main/scala/codegen/CppWriter/ArrayCppWriter.scala

```diff
@@ -172,6 +172,26 @@ case class ArrayCppWriter (
 
   private def getConstructorMembers: List[CppDoc.Class.Member] = {
     val defaultValues = getDefaultValues
+    // Only write this constructor if the array has more than one element
+    val singleElementConstructor =
+      if arraySize == 1 then None
```

Comment:
Should this be Nil and List(...) instead of None and Some(...). If this works, it looks like Scala may be implicitly converting None to Nil and Some(...) to List(...).

Reply:
I think that is what's happening, but Nil and List() does seem a little clearer